lib/include/kframework/avm/avm-configuration.md

```diff
@@ -25,35 +25,50 @@ module AVM-CONFIGURATION
       </returnstatus>
 
       // The transaction group as submitted
-      <txGroup/>
+      <transactions/>
 
       // Transaction can create inner transactions, and we chose to treat them similarly to the outer ones.
       // We add them into an execution deque --- inner transactions are executed right after their parent one.
       // Initially, the execution deque will contain the transactions from the submitted group (up to `MaxTxGroupSize`, 16 currently).
       // Outer transactions are referred to by their actual `txID`s. Inner transaction will be assigned "fake" `txID`s,
       // starting from `MaxTxGroupSize` (currently 16).
-      <txnDeque>
-        <deque>         .List </deque>
-        <dequeIndexSet> .Set  </dequeIndexSet>
-      </txnDeque>
-
-      // The execution context of the current transaction.
-      <currentTxnExecution>
-        // Globals are mostly immutable during the group execution,
-        // besides the application-related fields: CurrentApplicationID, CreatorID
-        // and CurrentApplicationAddress
-        <globals/>
-
-        // the `<teal>` cell will control evaluation of TEAL code of the current transaction.
-        // The semantics of TEAL has *read-only* access to the `<blockchain>` cell
-        // and *read-write* access to the `<effects>` cell.
-        <teal/>
-
-        // the effects of the transaction. Upon approval of the transaction,
-        // its effects will be applied onto the `<blockchain>` cell.
-        // TODO: how to represent effects? We need to track changes to accounts, assets and apps.
-        <effects> .List </effects>
-      </currentTxnExecution>
+      <avmExecution>
+
+        <currentTx> "0" </currentTx>
+
+        <txnDeque>
+          <deque>         .List </deque>
+          <dequeIndexSet> .Set  </dequeIndexSet>
+        </txnDeque>
+
+        // The execution context of the current transaction.
+        <currentTxnExecution>
+          // Globals are mostly immutable during the group execution,
+          // besides the application-related fields: CurrentApplicationID, CreatorID
+          // and CurrentApplicationAddress
+          <globals/>
+
+          // the `<teal>` cell will control evaluation of TEAL code of the current transaction.
+          // The semantics of TEAL has *read-only* access to the `<blockchain>` cell
+          // and *read-write* access to the `<effects>` cell.
+          <teal/>
+
+          // the effects of the transaction. Upon approval of the transaction,
+          // its effects will be applied onto the `<blockchain>` cell.
+          // TODO: how to represent effects? We need to track changes to accounts, assets and apps.
+          <effects> .List </effects>
+
+          <lastTxnGroupID> "" </lastTxnGroupID>
+
+        </currentTxnExecution>
+
+        <innerTransactions> .List </innerTransactions>
+
+        <activeApps> .Set </activeApps>
```

Comment:
```suggestion
        // Applications which are currently on the call stack. This cell is needed so that we can check for re-entrant
        // app calls. The `<txnDeque>` is not sufficient for this because it contains transactions that were not yet called but 
        // will be called further back in the call stack. 
        <activeApps> .Set </activeApps>
```